src/wp-admin/network/users.php

```diff
@@ -107,12 +118,27 @@
 
 							case 'notspam':
 								$user = get_userdata( $user_id );
+								if ( is_super_admin( $user->ID ) ) {
+									wp_die(
+										sprintf(
+											/* translators: %s: User login. */
+											__( 'The user %s is a network administrator and cannot be modified.' ),
+											esc_html( $user->user_login )
+										)
+									);
+								}
 
 								$userfunction = 'all_notspam';
-								$blogs        = get_blogs_of_user( $user_id, true );
 
-								foreach ( (array) $blogs as $details ) {
-									update_blog_status( $details->userblog_id, 'spam', '0' );
+								/** This filter is documented in wp-admin/network/users.php#L95 */
+								if ( apply_filters( 'network_user_spam_propagate_to_blogs', false, $user_id ) ) {
+									$blogs = get_blogs_of_user( $user_id, true );
+
+									foreach ( $blogs as $details ) {
```

Comment:
Might as well make this more concise:

```suggestion
									foreach ( get_blogs_of_user( $user_id, true ) as $details ) {
```